hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/storage/HoodieHFileWriter.java

```diff
@@ -122,7 +128,13 @@ public boolean canWrite() {
 
   @Override
   public void writeAvro(String recordKey, IndexedRecord object) throws IOException {
-    byte[] value = HoodieAvroUtils.avroToBytes((GenericRecord)object);
+    byte[] value = HoodieAvroUtils.avroToBytes((GenericRecord) object);
```

Comment:
why do the avroToBytes twice? better to check the condition first.?

Reply:
We should not empty/change the passed in record object 'key' field, else the caller will have the in-memory copy of the record object with key missing and affects all users of it. So, i need a copy of the record object, where i can empty the key field and then save to disk. The second de-serialization back to a new record object where i can change the field is needed. If there are any other better ways to doing this, happy to change.

Reply:
This is another way to implement this without perf issue:
byte[] value;
if (schemaRecordKeyField.isPresent()) {
    int keyFieldPos = this.schemaRecordKeyField.get().pos();
    Object origKey = object.get(keyFieldPos);
    object.put(keyFieldPos, StringUtils.EMPTY_STRING);
    value = HoodieAvroUtils.avroToBytes((GenericRecord)object);
    object.put(keyFieldPos, origKey);
} else {
   value = HoodieAvroUtils.avroToBytes((GenericRecord) object);
}

Reply:
Right, this is better than double serialization. Incorporated this.